### ISSUE TITLE
Wrong parsing of mowerState?

### ISSUE THREAD

**Issue by erikmol** [alistair23/AutoMower-BLE#13]
When my mower (305) was mowing, the mowerState (4586, 2) was returning the raw value of 6 which [is being parsed](https://github.com/alistair23/AutoMower-BLE/blob/34b3174f83d76b16ca25667608aa5d935ecae9e6/automower_ble/response.py#L153) as "checkSafety".

Decompiling the android apk however shows that 4586, 2 parsed with the enum `ProtocolTypes$IMowerAppState` which holds the following values
```
    IMOWERAPP_STATE_OFF(0),
    IMOWERAPP_STATE_WAIT_FOR_SAFETYPIN(1),
    IMOWERAPP_STATE_STOPPED(2),
    IMOWERAPP_STATE_FATAL_ERROR(3),
    IMOWERAPP_STATE_PENDING_START(4),
    IMOWERAPP_STATE_PAUSED(5),
    IMOWERAPP_STATE_IN_OPERATION(6),
    IMOWERAPP_STATE_RESTRICTED(7),
    IMOWERAPP_STATE_ERROR(8);
```
Here, the value 6 is parsed to "In operation" which was a better description of my mower. When parked the value 7 is returned which is a bit confusing.

Anyone else with the same issue?

**Comment by enicky** [comment]
> Anyone else with the same issue?

Yes I'm having the same issue with my Minimo. I also saw that the mappings were differently

**Comment by alistair23** [comment]
The mappings seem to track with my 305. So it's possible you have different mappings. The enum in the reverse engineered Husqvarna app doesn't seem to match what the actual mower does though, which makes the whole thing very confusing.

I'm all open to improvements, but I don't want to break the Husqvarna ones as they *seem* correct

**Comment by enicky** [comment]
Made a small [PR](https://github.com/alistair23/AutoMower-BLE/pull/24). This way it's backwards compatible and you can change the model to be able to use the different mapping.

My Gardena Minimo uses the different mapping.

**Comment by erikmol** [comment]
> The enum in the reverse engineered Husqvarna app doesn't seem to match what the actual mower does though, which makes the whole thing very confusing.
> 
> 
> 
> I'm all open to improvements, but I don't want to break the Husqvarna ones as they *seem* correct

So how did you map these originally? I.e. how did checkSafety map to "6"? Because that does not match with my Automower 305.

Perhaps we can compare if there are different enums in different android apps (Husqvarna, Gardena, Flymo, ...) to understand if we expect differences or not.

**Comment by enicky** [comment]
> Perhaps we can compare if there are different enums in different android apps (Husqvarna, Gardena, Flymo, ...) to understand if we expect differences or not.

That's a good idea. Looking at my Gardena app I see the following values for the enum : 
```
public enum IMowerAppState {
        IMOWERAPP_STATE_OFF(0),
        IMOWERAPP_STATE_WAIT_FOR_SAFETYPIN(1),
        IMOWERAPP_STATE_STOPPED(2),
        IMOWERAPP_STATE_FATAL_ERROR(3),
        IMOWERAPP_STATE_PENDING_START(4),
        IMOWERAPP_STATE_PAUSED(5),
        IMOWERAPP_STATE_IN_OPERATION(6),
        IMOWERAPP_STATE_RESTRICTED(7),
        IMOWERAPP_STATE_ERROR(8);

        .....
    }

```

**Comment by erikmol** [comment]
> > Perhaps we can compare if there are different enums in different android apps (Husqvarna, Gardena, Flymo, ...) to understand if we expect differences or not.
> 
> 
> 
> That's a good idea. Looking at my Gardena app I see the following values for the enum : 
> 
> ```

These are the same as I found in the Husqvarna app. So until we have this sorted out 100% and confirm that mappings are really different, I think #24 is obsolete.

**Comment by alistair23** [comment]
> So how did you map these originally? I.e. how did checkSafety map to "6"? Because that does not match with my Automower 305.

I basically just watched the mower and checked what it sent me. Originally I used the enum above, but from memory it gave me strange results.

A lot has changed since then, so it's possible there was some other issue that has since been fixed
